### PR TITLE
Keep rabl and implement Active Model Serializer as v2

### PR DESCRIPTION
**Pull request by yurifrl**
as @JDutil asked i merged it with master, and now, to organize myself i'm moving https://github.com/spree/spree/pull/6303 to here, there are some things missing, as @BenMorganIO  pointed, such as https://github.com/spree/spree/pull/6280

i'm also using this thread as a guide line https://github.com/spree/spree/pull/6033

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://azclick-ams.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://azclick-ams.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by yurifrl** [comment]
did i mess-up with this merge? it doesn't seem right.

**Comment by yurifrl** [comment]
i give up XD

File: api/spree_api.gemspec [hunk 1]
```diff
@@ -17,6 +17,7 @@ Gem::Specification.new do |gem|
   gem.version       = version
 
   gem.add_dependency 'spree_core', version
+  gem.add_dependency 'active_model_serializers', '0.9.3'
```
Comment (JDutil) [inline api/spree_api.gemspec]: I believe we want to go with ~> 0.8.x here as 0.10.x is being based off 0.8 not 0.9.

**Comment by JDutil** [comment]
@yurifrl nice start.  fyi https://github.com/bbatsov/rubocop can be used to find/fix many of the hound issues obviously tough dealing with all the spam from copy/pasted code :)

File: api/spree_api.gemspec (hunk 1 above)
Comment (BenMorganIO) [inline api/spree_api.gemspec]: @JDutil good catch.

Comment (yurifrl) [inline api/spree_api.gemspec]: agreed

**Comment by BenMorganIO** [comment]
I'm stilling going to sit on the "should be done in small batches" argument. I'm fearful of doing everything at once. One of the reasons was to be able to iron out mistakes before there were 7000 lines of code to edit. One mistake is that there's no clear way to implement `respond_with :xml` unless you do a lot of code deletion and editing.

What's also great about small things to look at instead large amounts is its easier to reason about the requirements such as the JSON API. We've spent a lot of time :+1:ing it but we need to spend some time analyzing the requirements and how this will impact Spree.

This large PR, if it doesn't get merged, a super awesome example of what could be the API. I'm :-1: on it and its implementation. @yurifrl we should work on setting up some hard requirements such as what JSON API implementations we need and how to structure large amounts of JSON for Ember Data. I think if we can make an outline of those two (because right now I don't fully know what a good JSON docment looks like for Ember Data and what the short comings are) then I think we could very much build the API  to a specification that a lot of people would enjoy with a more documented background.

**Comment by yurifrl** [comment]
i agree because of one reason, today i found out that ember does not support nested attributes by default :(

**Comment by BenMorganIO** [comment]
@yurifrl Can you help contribute to the RFC? https://github.com/BenMorganIO/rfcs/blob/v2-api/inactive/000-v2-api.md

**Comment by BenMorganIO** [comment]
I've got it started and it needs work. Motivation needs some comments. Not finished writing the detailed design yet. So much writing!

**Comment by yurifrl** [comment]
Sure, i do what i can.

**Pull request closed without merging**